### PR TITLE
Mask password with *******

### PR DESCRIPTION
**Pull request by harcher81**
And support of CloudBees AWS Credentials Plugin as credential provider.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -55,6 +55,16 @@
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>aws-credentials</artifactId>
```
Comment (jglick) [inline pom.xml]: Really do not want this dependency. If anything, `aws-credentials` could depend on the more generic `credentials-binding`. Anyway this is unrelated to the main thrust of the PR.

File: pom.xml [hunk 2]
```diff
@@ -55,6 +55,16 @@
...
+      <version>1.11</version>
+    </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>envinject</artifactId>
```
Comment (jglick) [inline pom.xml]: Why?

File: src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep.java [hunk 3]
```diff
@@ -148,7 +148,7 @@
                 @Override protected void eol(byte[] b, int len) throws IOException {
                     Matcher m = p.matcher(new String(b, 0, len));
                     if (m.find()) {
-                        logger.write(m.replaceAll("****").getBytes());
+                        logger.write(m.replaceAll("********").getBytes());
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep.java]: Why?

File: src/main/java/org/jenkinsci/plugins/credentialsbinding/MultiBinding.java [hunk 4]
```diff
@@ -97,17 +94,17 @@ public Unbinder getUnbinder() {
     /** Callback run at the end of a build. */
     public interface Unbinder extends Serializable {
         /** Performs any needed cleanup. */
-        void unbind(@Nonnull Run<?,?> build, FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException;
+        void unbind(@Nonnull Run<?,?> build, FilePath workspace) throws IOException, InterruptedException;
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/credentialsbinding/MultiBinding.java]: Incompatible.

**Comment by jglick** [comment]
Seems like #12 is already covering this?

**Comment by harcher81** [comment]
Yes, It coverages the same problems, but I think, this is a better solution. In a "Username and password (conjoined)" binding step, if you inject Foo:Bar, with this fix, this hide all occurrences of Bar and not just Foo:bar.

**Comment by jglick** [comment]
> In a "Username and password (conjoined)" binding step, if you inject Foo:Bar, with this fix, this hide all occurrences of Bar and not just Foo:bar

OK…but the plugin was not setting any variable to `Bar` to begin with. It was only setting `Foo:Bar`, which we mask in case that was passed e.g. to `curl -u $USERPASS …` and the shell was in echo mode. Of course some poorly implemented command might be printing the password later for some reason, but if this is a problem you need merely switch to `UsernamePasswordMultiBinding`.
